sycl/include/CL/sycl/detail/group_sort_impl.hpp

```diff
@@ -0,0 +1,224 @@
+//==------------ group_sort_impl.hpp ---------------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// This file includes some functions for group sorting algorithm implementations
+//
+
+#pragma once
+
+#include <CL/sycl/detail/helpers.hpp>
+
+#ifdef __SYCL_DEVICE_ONLY__
+#include "group_sort_util.hpp"
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace detail {
+
+// ---- merge sort implementation
+
+// following to functions could be useless if std::[lower|upper]_bound worked
+// well
+template <typename Acc, typename Value, typename Compare>
+std::size_t my_lower_bound(Acc acc, std::size_t first, std::size_t last,
```

Comment:
Is there any reason to prefix these with `my_` if they're already in `sycl::detail::`?

Same question for `my_upper_bound`, `my_swap` and so on.

Reply:
No reason. Just to not mix with `std::` functions

Reply:
For such cases we simply use namespaces like `detail::upper_bound` or `std::upper_bound`.